### DISCUSSION TITLE
pkg-config not searching /opt/homebrew/lib/pkgconfig on Apple Silicon

### DISCUSSION THREAD

**Discussion by benjaminmordaunt** [Homebrew/discussions#2987] [Everyday usage]
`pkg-config` fails to find libraries installed by Homebrew, so I decided to investigate.

`pkg-config --variable pc_path pkg-config` gives `/opt/local/lib/pkgconfig:/opt/local/share/pkgconfig`. I would expect to see the expanded version of `$HOMEBREW_PREFIX/lib/pkgconfig` in here too, which for my M1 system is `/opt/homebrew/lib/pkgconfig`. In fact, inspecting pkg-config.rb, it does appear that this should be in pc_path.

At the moment, I am having to add this to my `.zshrc`:

```sh
# Add seemingly missing pkg-config path? Not sure why this is happening...
export PKG_CONFIG_PATH="$HOMEBREW_PREFIX/lib/pkgconfig"
```

Any idea why this is happening? I have tried reinstalling pkg-config, so please don't suggest that :P

**Comment by carlocab** [comment]
What's the output of `which -a pkg-config`?

**Reply by benjaminmordaunt** [reply]
`which -a pkg-config`:
```
/opt/local/bin/pkg-config
/opt/homebrew/bin/pkg-config
/opt/local/bin/pkg-config
/opt/homebrew/bin/pkg-config
```

`which pkg-config`:
```
/opt/local/bin/pkg-config
```

I guess that's the issue :P How to fix?

**Reply by carlocab** [reply]
That's your problem, no? You have a different (probably MacPorts) `pkg-config` in `/opt/local/bin`, and `/opt/local/bin` comes earlier in your `PATH` than `/opt/homebrew/bin`.

If you want to use Homebrew `pkg-config`, then you need to either call it as `/opt/homebrew/bin/pkg-config` or make sure `/opt/homebrew/bin` comes earlier in your `PATH` than `/opt/local/bin`.

**Reply by benjaminmordaunt** [reply]
Cool, thanks. Forgot I ever used macports on this system and that does seem to be the issue.

**Reply by tex0l** [reply]
You saved my life
